Update caps in decoder when receiving key frames

jacek-manko-red opened the pull request:
The resolution in WebRTC does change
Update caps when _encodeWidth and height are non-zero
Those caps are also used by GStreamerVideoFrameLibWebRTC and later by LibWebRTCStatsCollector.

Without this change, resolution in inbound-rtp statistics is never updated

philn commented on Source/WebCore/platform/mediastream/libwebrtc/gstreamer/GStreamerVideoDecoderFactory.cpp:
```diff
@@ -212,6 +212,11 @@ class GStreamerWebRTCVideoDecoder : public webrtc::VideoDecoder {
             return WEBRTC_VIDEO_CODEC_UNINITIALIZED;
         }
 
+        if (inputImage._encodedWidth > 0 && inputImage._encodedHeight > 0) {
```
Would it make sense to keep track of the input image dimensions and update the caps only when those dimensions change?

With the current version of this patch I'm a bit worried with the extra allocations.

jacek-manko-red replied:
Yes, of course. I've updated the branch

philn replied:
Thanks! I can take care of upstreaming and backporting of this PR.

jacek-manko-red committed:
Update caps in decoder when receiving key frames

The resolution in WebRTC does change
Update caps when _encodeWidth and height are non-zero and have changed
Those caps are also used by GStreamerVideoFrameLibWebRTC
and later by LibWebRTCStatsCollector.

Without this change, resolution in inbound-rtp statistics is never updated

philn commented:
Upstreamed as https://commits.webkit.org/282115@main
Backported to wpe-2.38: ef5a746bb7dba954fbe52fe384689f499a53e450
